mautic/mautic
# fix: lookup contact by ip in mtc.js tracker *except* if anonymised

see https://forum.mautic.org/t/mtc-js-ip-address-uniqueness-questions/20632

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for all features, enhancements and bug fixes (until 3.3.0 is released) 
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



See https://forum.mautic.org/t/mtc-js-ip-address-uniqueness-questions/20632 for a discussion.

In the mtc.js tracker, the intent is to lookup the contact by IP *except* if anonymised. 
Instead, it was and'ing the two.

#### Description:


#### Steps to test this PR:
Setup site using mtc.js tracking. Enable `Identify visitors by IP`, disable `Anonymize IP`.
Open incognito browser, go to site, observe that anonymous user is created by your ip.
do something to post your email (e.g. post form). Close.
Open new incognito browser, go to site, observe that a new anonymous user is created instead of adopting previous.
